src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -71,9 +77,28 @@ CHIP_ERROR AES_CCM_encrypt(const uint8_t * plaintext, size_t plaintext_length, c
 
     const psa_algorithm_t algorithm = PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CCM, tag_length);
     psa_status_t status             = PSA_SUCCESS;
-    psa_aead_operation_t operation  = PSA_AEAD_OPERATION_INIT;
     size_t out_length               = 0;
-    size_t tag_out_length           = 0;
+
+#if CHIP_CRYPTO_PSA_AEAD_SINGLE_PART
+    // Temporary buffer is required since during encryption plaintext and tag are not contiguous in memory.
```

Comment:
```suggestion
    // Temporary buffer is required since during encryption plaintext and tag are decrypted as one unit
    // and then encrypted part and tag are separated out. 
    // ciphertext is only guranteed to have `plaintext_length` size and we need a buffer of
    // plaintext_length + tag_length for output
```

Reply:
@LuDuda is this more accurate? Trying to make sure I understand.

Reply:
Thanks, I updated both comments according to your suggestions.